File: src/main/java/no/fintlabs/repository/filters/InstanceStorageStatusQueryFilter.java
package no.fintlabs.repository.filters;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.Collection;

@Getter
@AllArgsConstructor
public class InstanceStorageStatusQueryFilter {

    public static final InstanceStorageStatusQueryFilter EMPTY =
            new InstanceStorageStatusQueryFilter(null, null);

    private final Collection<String> instanceStorageStatusNames;
    private final Boolean neverStored;
}

File: src/main/kotlin/no/novari/flyt/history/repository/filters/InstanceStorageStatusQueryFilter.kt
package no.novari.flyt.history.repository.filters

data class InstanceStorageStatusQueryFilter(
    val instanceStorageStatusNames: Collection<String>? = null,
    val neverStored: Boolean? = null,
) {
    companion object {
        @JvmField
        val EMPTY = InstanceStorageStatusQueryFilter()
    }
}
